src/main/java/uk/ac/cam/cl/dtg/teaching/pottery/api/RepoController.java

```diff
@@ -57,9 +59,12 @@ RepoInfo makeRepo(
       @FormParam("taskId") String taskId,
       @FormParam("usingTestingVersion") Boolean usingTestingVersion,
       @FormParam("validityMinutes") Integer validityMinutes,
-      @FormParam("variant") String variant)
+      @FormParam("variant") String variant,
+      @FormParam("seed") Integer seed,
+      @FormParam("extraParameters") String extraParameters)
```

Comment:
Is there not a way to tell jaxrs that this is a list of key/value pairs?

Reply:
So, extraParameters can carry arbitrary JSON, not just string->string (for reasons below).
Given this project doesn't currently include Jackson, I thought it best to just make this a String.

Reply:
Actually, of course it includes Jackson. I don't know if it would work as an `@FormParam` but I'll try it

Reply:
Nah, it doesn't like it. Unless you want to restrict the extraParameters to string-string pairs? I think a `Map<String,String>` would be okay. But then you won't be able to do the same stuff as with parameterisation.

Reply:
The benefit of having more than string-string pairs being that they get passed to the template and so it might be useful to have mode complex types?

Reply:
Right.